src/arrayMethodReduce.js

```diff
@@ -3,9 +3,26 @@
 /**
  * Implement method Reduce
  */
+
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc = initialValue;
+    let startIndex = 0;
+
+    if (arguments.length === 1) {
+      acc = this[0];
+      startIndex = 1;
+    } else if (arguments.length > 1 && initialValue === undefined) {
+      acc = 'undefined';
```

Comment:
why do you need `acc = 'undefined'`?

just check if arguments.length < 2  and do these actions
`acc = this[0];
startIndex = 1;`